pxr/usd/lib/usd/stage.h

```diff
@@ -700,6 +700,33 @@ class UsdStage : public TfRefBase, public TfWeakBase {
     USD_API
     UsdPrim GetPrimAtPath(const SdfPath &path) const;
 
+    /// Return the UsdObject at \p path, or an invalid UsdObject if none exists.
+    ///
+    /// If \p path indicates a prim beneath an instance, returns an instance
+    /// proxy prim if a prim exists at the corresponding path in that instance's 
+    /// master.
+    ///
+    /// Suggested use:
+    ///
+    /// \code
+    /// if (UsdObject myObj = stage.GetObjectAtPath(path)) {
```

Comment:
We can just use UsdObject::As in this example, since As is documented to return an invalid object if it's called on an invalid object, like:

```
if (UsdObject obj = stage->GetObjectAtPath(path)) {
    if (UsdPrim prim = obj.As<UsdPrim>()) {
        // Do things with prim
    }
    else if (UsdProperty prop = obj.As<UsdProperty>()) {
        // Do things with property. We can also cast to
        // UsdRelationship or UsdAttribute using this same pattern.
    }
}
else {
    // No object at specified path
}
```